.github/workflows/test.yml

```diff
@@ -9,6 +9,23 @@ on:
       - '.gitignore'
 
 jobs:
+  tests:
+    runs-on: ubuntu-latest
+    container:
+      image: python:3.12
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          pip install -r requirements.txt
+
+      - name: Run tests
+        run: |
+          python -m unittest discover -s otterai/tests -p 'test_*.py'
```

Comment:
✅ Running tests with unittest is a solid approach! Make sure to cover edge cases in your test suite.